lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -1769,6 +1770,109 @@ static GraphOperationInst *tryToPromoteTensorFromScalars(
   return result;
 }
 
+/// This function transform the apply of the "__tf_tensor_from_scalar" function
+/// to a graph node (graph_op inst).
+///
+/// Similar to tryToPromoteTensorFromScalars(), the resulting graph node can be
+/// a "const", if its input scalar operand is a compile-time constant.
+/// Different from tryToPromoteTensorFromScalars(), we can also transform the
+/// apply into a "tfc.scalarToTensor" graph_op, which the partition pass then
+/// uses to do scalar promotion. See the comment above "func
+/// _TFTensorFromScalar<Scalar>" in Tensor.swift for more details.
+static GraphOperationInst *
+transformTensorFromScalar(ApplyInst *apply,
```

Comment:
Instead of adding a new function, would it make sense to refactor tryToPromoteTensorFromScalars to create tfc.scalarToTensor op for 0-d case?

Reply:
I don't think so. The amount of overlapping code is not a lot, so unifying would add code complexity.

Reply:
Sounds good.